File: src/Image4ioSDK/JSON.vb
Imports Newtonsoft.Json
Imports System.ComponentModel

Namespace JSON


    Public Enum ResultsStatusEnum
        Uploaded
        fetched
        Copied
        Moved
        Deleted
        Created
        AlreadyExist
        Overwrited
        [Error]
    End Enum

    Public Class JSON_Error
        <JsonProperty("message")> Public Property _ErrorMessage As String
        <JsonProperty("statusCode")> Public Property _ErrorCode As Integer
    End Class


#Region "JSON_ImageMetadata"
    Public Class JSON_ImageMetadata
        'https://stackoverflow.com/posts/43715009/revisions
        <JsonProperty("userGivenName")>
        Public Property Filename As String
        <JsonProperty("orginal_name")>
        Private WriteOnly Property Filename2 As String
            Set(ByVal value As String)
                Filename = value
            End Set
        End Property

        Public Property folder As String ''list only, not in single metadata
        <JsonProperty("name")> Public Property Path As String
        Public Property size As Integer
        Public Property format As String
        Public Property width As Integer
        Public Property height As Integer
        Public Property createdAtUTC As Date
        Public Property updatedAtUTC As Date

        Public ReadOnly Property DirectUrl As String
            Get
                Return String.Format("https://cdn.image4.io/{0}/f_auto{1}", CloudName, Path)
            End Get
        End Property
    End Class
#End Region

#Region "JSON_Upload"
    Public Class JSON_MultiUpload
        Public Property uploadedFiles As List(Of JSON_Uploadedfile)
    End Class
    Public Class JSON_Uploadedfile
        Public Property orginal_name As String
        Public Property name As String
        Public Property status As ResultsStatusEnum
    End Class
#End Region

#Region "JSON_FolderList"
    Public Class JSON_FolderList
        Public Property folders As List(Of JSON_FolderMetadata)
        Public Property files As List(Of JSON_ImageMetadata)
    End Class
    Public Class JSON_FolderMetadata
        Public Property name As String
        Public Property parent As String
    End Class
#End Region



End Namespace

File: Image4ioSDK/Image4ioSDK/JSON.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
//using Image4ioSDK;

namespace Image4ioSDK.JSON
{
    public enum ResultsStatusEnum { Uploaded, fetched, Copied, Moved, Deleted, Created, AlreadyExist, Overwrited, Error }


    public class JSON_Error
    {
        [JsonProperty("message")]
        public string _ErrorMessage { get; set; }
        [JsonProperty("statusCode")]
        public int _ErrorCode { get; set; }
    }


    #region "JSON_ImageMetadata"
    public class JSON_ImageMetadata
    {
        //'https://stackoverflow.com/posts/43715009/revisions
        [JsonProperty("userGivenName")]
        public string Filename { get; set; }

        [JsonProperty("orginal_name")]
        public string Filename2
        {
            set
            {
                Filename = value;
            }
        }

        public string folder { get; set; }//''list only, not in single metadata
        [JsonProperty("name")]
        public string Path { get; set; }
        public int size { get; set; }
        public string format { get; set; }
        public int width { get; set; }
        public int height { get; set; }
        public string createdAtUTC { get; set; }
        public string updatedAtUTC { get; set; }
        public string DirectUrl
        {
            get { return String.Format("https://cdn.image4.io/{0}/f_auto{1}", Image4ioSDK.Basic.CloudName, Path); }
        }

    }
    #endregion


    #region "JSON_Upload"
    public class JSON_MultiUpload
    {
        public List<JSON_Uploadedfile> uploadedFiles;
    }
    public class JSON_Uploadedfile
    {
        public string orginal_name;
        public string name;
        public ResultsStatusEnum status;
    }
    #endregion

    #region "JSON_FolderList"
    public class JSON_FolderList
    {
        public List<JSON_FolderMetadata> folders { get; set; }

        public List<JSON_ImageMetadata> files { get; set; }
    }
    public class JSON_FolderMetadata
    {
        public string name;
        public string parent;
    }
    #endregion






}
